sim-lib/src/lib.rs

```diff
@@ -275,6 +277,44 @@ impl Simulation {
         Ok(())
     }
 
+    // validate_node_network validates that the nodes that are on the same bitcoin network.
+    // e.g. node A and B are on testnet, node C is on mainnet should return an error.
+    async fn validate_node_network(&self) -> Result<(), LightningError> {
+        let mut networks = Vec::new();
+        let mainnet = Network::Bitcoin.to_string().to_lowercase();
+
+        for node in self.nodes.values() {
```

Comment:
if each node info declares just one `network`, maybe you could replace the validation logic below with something like

```rs

let mainnet = Network::Bitcoin.to_string().to_lowercase();
let mut valid_network = Option::None;

for node in self.nodes.values() {
    let network = node.lock().await.get_info().network;

    if network == mainnet {
        return Err(LightningError::ValidationError(
            "mainnet is not supported".to_string(),
        ));
    }

    let valid_replaced = valid_network.replace(network.clone());

    if valid_replaced.is_some() && valid_replaced != network {
        return Err(LightningError::ValidationError(format!(
            "nodes are not on the same network {}",
            network,
        )));
    }
}

Ok(())
```